src/mlpack/bindings/R/mlpack/tests/testthat/test-object_type.R

```diff
@@ -0,0 +1,10 @@
+test_that("Test for mlpack S3 class ", {
+
+    set.seed(1234)
+    x <- matrix(rnorm(10*5), ncol = 5)
+
+    res <- mlpack::knn(query = x, reference = x, k = 3)
+
+    expect_s3_class(res, c("mlpack_knn", "mlpack_model_binding"))
+
+})
```

Comment:
Any particular reason not to put this in `test-R_binding.R`?  I think that this will actually not be included in the distributed tests unless you modify `R_TESTTHAT_SOURCES` in `src/mlpack/bindings/R/CMakeLists.txt`; so I really wonder whether it is easier to just put it in the other file.  (Sorry, the overhead of the binding generator might be a little frustrating, but such is life.)

Reply:
Purely personal preference to not overload an existing test file. This struck me as something new.  But this is more than 100% the old 'six of one, half a dozen of another' as the test runner runs both.

Or not. I just realized that I didn't update CMakeLists.txt.  This is after all an overly complicated non-standard _meta_ package creating an R package, and it fooled me again.  So I will move it into the one existing file `test-R_binding.R`.